sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -27,16 +27,27 @@ import org.apache.spark.sql.execution.{BinaryNode, SparkPlan}
  * :: DeveloperApi ::
  */
 @DeveloperApi
-case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNode {
+case class CartesianProduct(
+    left: SparkPlan,
+    right: SparkPlan,
+    buildSide: BuildSide) extends BinaryNode {
   override def output: Seq[Attribute] = left.output ++ right.output
 
+  private val (small, big) = buildSide match {
+    case BuildRight => (left, right)
```

Comment:
In other places, `BuildRight` means the right side is a small table, such as `org.apache.spark.sql.execution.joins.HashJoin`, and we usually broadcast it. Could you follow this semantics?